### PR TITLE
MGMT-4088 add nmstate package to the AssistedService image

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
Add build and setup of the nmstate lib to the Assisted Service container
Currently we are building and setting the code from the master branch
Once the needed code is pushed into nmstate 4.8 reelase, we will be
able to installed released package

### CODE REVIEW

**Comment by filanov** [comment]
How much does it increase the image size? (just to know)

**Comment by filanov** [comment]
@YuviGold @gamli75 does it require anything in downsteam?

**Comment by yevgeny-shnaidman** [comment]
@filanov it doubles it, but once we move to use the release image, we won't need develop packages, so it will be smaller

**Comment by gamli75** [comment]
@filanov @yevgeny-shnaidman We can't install something from the internet (https://github.com/nmstate/nmstate) in the DS images. 
We need to check it with the on-prem guys - @djzager ?

**Comment by yevgeny-shnaidman** [comment]
> @filanov @yevgeny-shnaidman We can't install something from the internet (https://github.com/nmstate/nmstate) in the DS images.
> We need to check it with the on-prem guys - @djzager ?

@gamli75 this is currently a temp fix so that we can continue developing static network configuration. Eventually , what will be installed is an official release package of nmstate.

**Comment by filanov** [comment]
> > @filanov @yevgeny-shnaidman We can't install something from the internet (https://github.com/nmstate/nmstate) in the DS images.
> > We need to check it with the on-prem guys - @djzager ?
> 
> @gamli75 this is currently a temp fix so that we can continue developing static network configuration. Eventually , what will be installed is an official release package of nmstate.

When will it happen? Will it block us from deploying to production?

**Comment by gamli75** [comment]
> > > @filanov @yevgeny-shnaidman We can't install something from the internet (https://github.com/nmstate/nmstate) in the DS images.
> > > We need to check it with the on-prem guys - @djzager ?
> > 
> > 
> > @gamli75 this is currently a temp fix so that we can continue developing static network configuration. Eventually , what will be installed is an official release package of nmstate.
> 
> When will it happen? Will it block us from deploying to production?

In production, we are not using DS image for assisted-service so it should not block us. 
let's wait for @djzager answer.

**Comment by yevgeny-shnaidman** [comment]
> > > @filanov @yevgeny-shnaidman We can't install something from the internet (https://github.com/nmstate/nmstate) in the DS images.
> > > We need to check it with the on-prem guys - @djzager ?
> > 
> > 
> > @gamli75 this is currently a temp fix so that we can continue developing static network configuration. Eventually , what will be installed is an official release package of nmstate.
> 
> When will it happen? Will it block us from deploying to production?

i am not sure when their release is due, but i guess it is a matter of a coupe of weeks, since we needed to hurry with the PR in order for it to make to the next release

**Comment by YuviGold** [comment]
@yevgeny-shnaidman Can you update the PR title to have a summary? Because this what would appear on the git commit summary after the merge.
iirc it must be changed on d/s, but @djzager knows better.
In addition, can those "install" steps occur on a different stage before the final one? So only the nm package would be copied like we do with the other outputs.

**Comment by yevgeny-shnaidman** [comment]
> @yevgeny-shnaidman Can you update the PR title to have a summary? Because this what would appear on the git commit summary after the merge.
> iirc it must be changed on d/s, but @djzager knows better.
> In addition, can those "install" steps occur on a different stage before the final one? So only the nm package would be copied like we do with the other outputs.

@YuviGold what exactly do you want me to add to PR title? The update regarding not using this for DS?
also, since this is a python package, i cannot just copy it and that's it, i will have to check which additional packages it needs for running. I think it not worth the effort, since i will be updating the dockerfile to install an official nmstate release package shortly (once the release is ready), without any need to import the code, build and install it

**Comment by YuviGold** [comment]
> > @yevgeny-shnaidman Can you update the PR title to have a summary? Because this what would appear on the git commit summary after the merge.
> > iirc it must be changed on d/s, but @djzager knows better.
> > In addition, can those "install" steps occur on a different stage before the final one? So only the nm package would be copied like we do with the other outputs.
> 
> @YuviGold what exactly do you want me to add to PR title? The update regarding not using this for DS?
> also, since this is a python package, i cannot just copy it and that's it, i will have to check which additional packages it needs for running. I think it not worth the effort, since i will be updating the dockerfile to install an official nmstate release package shortly (once the release is ready), without any need to import the code, build and install it

Something like "Add nmstate to service image"
Regarding the Dockerfile, if you think it is going to get changed soon so I agree it doesn't worth it.

**Comment by yevgeny-shnaidman** [comment]
> > > @yevgeny-shnaidman Can you update the PR title to have a summary? Because this what would appear on the git commit summary after the merge.
> > > iirc it must be changed on d/s, but @djzager knows better.
> > > In addition, can those "install" steps occur on a different stage before the final one? So only the nm package would be copied like we do with the other outputs.
> > 
> > 
> > @YuviGold what exactly do you want me to add to PR title? The update regarding not using this for DS?
> > also, since this is a python package, i cannot just copy it and that's it, i will have to check which additional packages it needs for running. I think it not worth the effort, since i will be updating the dockerfile to install an official nmstate release package shortly (once the release is ready), without any need to import the code, build and install it
> 
> Something like "Add nmstate to service image"
> Regarding the Dockerfile, if you think it is going to get changed soon so I agree it doesn't worth it.

Isn't it what it says: "Add build and setup of the nmstate lib to the Assisted Service container
Currently we are building and setting the code from the master branch
Once the needed code is pushed into nmstate 4.8 reelase, we will be
able to installed released package" ?

**Comment by yevgeny-shnaidman** [comment]
> > > > @yevgeny-shnaidman Can you update the PR title to have a summary? Because this what would appear on the git commit summary after the merge.
> > > > iirc it must be changed on d/s, but @djzager knows better.
> > > > In addition, can those "install" steps occur on a different stage before the final one? So only the nm package would be copied like we do with the other outputs.
> > > 
> > > 
> > > @YuviGold what exactly do you want me to add to PR title? The update regarding not using this for DS?
> > > also, since this is a python package, i cannot just copy it and that's it, i will have to check which additional packages it needs for running. I think it not worth the effort, since i will be updating the dockerfile to install an official nmstate release package shortly (once the release is ready), without any need to import the code, build and install it
> > 
> > 
> > Something like "Add nmstate to service image"
> > Regarding the Dockerfile, if you think it is going to get changed soon so I agree it doesn't worth it.
> 
> Isn't it what it says: "Add build and setup of the nmstate lib to the Assisted Service container
> Currently we are building and setting the code from the master branch
> Once the needed code is pushed into nmstate 4.8 reelase, we will be
> able to installed released package" ?

done

**Comment by djzager** [comment]
> @filanov it doubles it, but once we move to use the release image, we won't need develop packages, so it will be smaller

is it possible to use a separate build stage in the Dockerfile to build nmstate from master? Sometimes it's prohibitively difficult but figured I should mention it.

> @filanov @yevgeny-shnaidman We can't install something from the internet (https://github.com/nmstate/nmstate) in the DS images.
> We need to check it with the on-prem guys - @djzager ?

@celebdor may know better, but I suspect that nmstate is already available d/s and should be updated after the 4.8 release.

**Comment by YuviGold** [comment]
> > > > @yevgeny-shnaidman Can you update the PR title to have a summary? Because this what would appear on the git commit summary after the merge.
> > > > iirc it must be changed on d/s, but @djzager knows better.
> > > > In addition, can those "install" steps occur on a different stage before the final one? So only the nm package would be copied like we do with the other outputs.
> > > 
> > > 
> > > @YuviGold what exactly do you want me to add to PR title? The update regarding not using this for DS?
> > > also, since this is a python package, i cannot just copy it and that's it, i will have to check which additional packages it needs for running. I think it not worth the effort, since i will be updating the dockerfile to install an official nmstate release package shortly (once the release is ready), without any need to import the code, build and install it
> > 
> > 
> > Something like "Add nmstate to service image"
> > Regarding the Dockerfile, if you think it is going to get changed soon so I agree it doesn't worth it.
> 
> Isn't it what it says: "Add build and setup of the nmstate lib to the Assisted Service container
> Currently we are building and setting the code from the master branch
> Once the needed code is pushed into nmstate 4.8 reelase, we will be
> able to installed released package" ?

It does. But when Prow bot merges a PR, it stashes all the PR commits and creates a new merge commit.
That merge commit's summary is being taken from the PR title and the body is a concatenation of all the PR commits' messages.

**Comment by yevgeny-shnaidman** [comment]
> > > > > @yevgeny-shnaidman Can you update the PR title to have a summary? Because this what would appear on the git commit summary after the merge.
> > > > > iirc it must be changed on d/s, but @djzager knows better.
> > > > > In addition, can those "install" steps occur on a different stage before the final one? So only the nm package would be copied like we do with the other outputs.
> > > > 
> > > > 
> > > > @YuviGold what exactly do you want me to add to PR title? The update regarding not using this for DS?
> > > > also, since this is a python package, i cannot just copy it and that's it, i will have to check which additional packages it needs for running. I think it not worth the effort, since i will be updating the dockerfile to install an official nmstate release package shortly (once the release is ready), without any need to import the code, build and install it
> > > 
> > > 
> > > Something like "Add nmstate to service image"
> > > Regarding the Dockerfile, if you think it is going to get changed soon so I agree it doesn't worth it.
> > 
> > 
> > Isn't it what it says: "Add build and setup of the nmstate lib to the Assisted Service container
> > Currently we are building and setting the code from the master branch
> > Once the needed code is pushed into nmstate 4.8 reelase, we will be
> > able to installed released package" ?
> 
> It does. But when Prow bot merges a PR, it stashes all the PR commits and creates a new merge commit.
> That merge commit's summary is being taken from the PR title and the body is a concatenation of all the PR commits' messages.

Fixed

File: Dockerfile.assisted-service
Comment (masayag) [inline Dockerfile.assisted-service:61]: https://github.com/nmstate/nmstate/releases/tag/v1.0.2 was released yesterday.
It can be downloaded from https://github.com/nmstate/nmstate/releases/download/v1.0.2/nmstate-1.0.2.tar.gz, extracted and run installed.

It will spare the git rpm and will allow using a fixated version for now (unless we need to rely on master)

Comment (yevgeny-shnaidman) [reply Dockerfile.assisted-service]: Done

File: Dockerfile.assisted-service
Comment (masayag) [inline Dockerfile.assisted-service:38]: can you omit the 'git' package ?

File: Dockerfile.assisted-service
Comment (romfreiman) [inline Dockerfile.assisted-service:39]: this is how it's being installed for CNV?

https://github.com/nmstate/kubernetes-nmstate/blob/master/build/Dockerfile#L4

what am I missing?

Comment (yevgeny-shnaidman) [reply Dockerfile.assisted-service:39]: no, for cnv it is being installed from official rpm package, we don't have it yet

File: Dockerfile.assisted-service
Comment (yevgeny-shnaidman) [reply Dockerfile.assisted-service]: Fixed

**Review by filanov — APPROVED** [review]

**Pull request merged**
